src/net_processing.cpp

```diff
@@ -2913,8 +2931,9 @@ void PeerManagerImpl::ProcessMessage(CNode& pfrom, const std::string& msg_type,
         // block-relay-only peer
         bool fBlocksOnly = m_ignore_incoming_txs || (pfrom.m_tx_relay == nullptr);
 
-        // Allow peers with relay permission to send data other than blocks in blocks only mode
-        if (pfrom.HasPermission(NetPermissionFlags::Relay)) {
+        // Allow peers with relay permission to send data other than blocks in blocks only mode, and if
```

Comment:
This sentence is difficult to understand. Perhaps "If this peer has relay permissions and we have a mempool, allow transactions announcements."